### ISSUE TITLE
iOS 13 MDCButton attribute title layout issue

### ISSUE THREAD

**Issue by nastasiupta** [material-components/material-components-ios#8561]
I just use the basic method button.setAttributedTitle(title, for: .normal)
But on iOS 13.0+ the button title is not displayed full.
On iOS 12 everything is fine

## Reproduction steps
use button.setAttributedTitle(title, for: .normal)
(SDKs MaterialComponents 83.0+)

## Expected behavior
Display full attributed title of the button

## Actual behavior
The title is not displayed in full. It adds ... in the middle of the attributed title.

A clear and concise description of what actually happened. Include screenshots or gifs where possible.

## Platform (please complete the following information)

 - Device: All iPhones
 - OS: iOS 13.0+
- SDK version: 92.3.1

## Additional context
iPhone 11 Pro iOS 13.0: https://drive.google.com/open?id=1hARLhiF56mTqTrNuAPkJfCmlMbT-3A46
iPhone Xs iOS 12.2:  https://drive.google.com/open?id=1_BPr4Pm52ZcoDxdoxFKie8IuXrX5Sodm

**Comment by romoore** [comment]
Have you tried calling `sizeToFit:` for manual layout or removing any width/leading/trailing constraints for Auto Layout?

**Comment by nastasiupta** [comment]
@romoore Yes, I just tried this right now after I set the attributed title and I still have the same issue

**Comment by romoore** [comment]
Thanks for confirming, @nastasiupta!  If you're using Auto Layout, can you try setting the `contentCompressionResistancePriority` to `.Required` for `.Horizontal`?

**Comment by nastasiupta** [comment]
@romoore  same issue...

**Comment by romoore** [comment]
Thanks for confirming again and for your help debugging!

**Comment by chandler7k** [comment]
Same situation..... any solution ?

**Comment by nastasiupta** [comment]
I found the issue... When I set the attributed title, it is an NSMutableAttributedString like this:
```
            let title = NSMutableAttributedString(attributedString: NSAttributedString(string: ButtonTitle.signIn.string, attributes: boldAttributes))
            title.append(NSAttributedString(string: " "))
            title.append(NSMutableAttributedString(attributedString: NSAttributedString(string: ButtonTitle.with.string, attributes: regularAttributes)))
            title.append(NSAttributedString(string: source, attributes: boldAttributes))
```

The issue is that I don't have any attributes for the white space between "Sign in" and "with". After I added, everything works fine:
```
            let title = NSMutableAttributedString(attributedString: NSAttributedString(string: ButtonTitle.signIn.string, attributes: boldAttributes))
            title.append(NSAttributedString(string: " ", attributes: boldAttributes))
            title.append(NSMutableAttributedString(attributedString: NSAttributedString(string: ButtonTitle.with.string, attributes: regularAttributes)))
            title.append(NSAttributedString(string: source, attributes: boldAttributes))
```

**Comment by romoore** [comment]
Thanks for the update, @nastasiupta.  Good to know these sharp edges exist with attributed strings!  Is this issue unique to MDCButton or is it reproducible with UIButton as well?

**Comment by ajayappledeveloper** [comment]
Thanks for the update.
